### ISSUE TITLE
使用mjsynth训练rec时出错

### ISSUE THREAD

**Issue by eiffelmen** [WenmuZhou/PytorchOCR#85]
## 首先我使用[您提供的脚本](https://github.com/WenmuZhou/OCR_DataSet/blob/master/convert/rec/mjsyhtn2txt.py)完成数据集的转换，转换后的数据集如下：
`mnt/ramdisk/max/90kDICT32px/2697/6/466_MONIKER_49537.jpg	MONIKER
  mnt/ramdisk/max/90kDICT32px/2697/6/465_Ecclesiastics_24500.jpg	Ecclesiastics `

## 我使用转换后数据集训练文本识别模型，执行如下命令，会出现随机的段错误或者内存错误：
`python ./tools/rec_train.py --config ./config/rec_train_config_self.py`

## 其中rec_train_config_self.py的部分内容如下（我只修改了字典、训练集、验证集的路径）：
`
config.loss = {
    'type': 'CTCLoss',
    'blank_idx': 0,
}
config.dataset = {
    # 'alphabet': r'path/dic.txt',
    'alphabet': r'torchocr/datasets/alphabets/enAlphaNumPunc90.txt',
    # 'alphabet': r'torchocr/datasets/alphabets/digit.txt',
    'train': {
        'dataset': {
            'type': 'RecTextLineDataset',
            'file': r'path/mnt/ramdisk/max/90kDICT32px/annotation_train_other.txt',
            'input_h': 32,
            'mean': 0.5,
            'std': 0.5,
            'augmentation': False,
        },
        'loader': {
            'type': 'DataLoader',  # 使用torch dataloader只需要改为 DataLoader
            'batch_size': 4,
            'shuffle': True,
            'num_workers': 1,
            'collate_fn': {
                'type': 'RecCollateFn',
                'img_w': 120
            }
        }
    },
    'eval': {
        'dataset': {
            'type': 'RecTextLineDataset',
            'file': r'path/mnt/ramdisk/max/90kDICT32px/annotation_val_other.txt',
            'input_h': 32,
            'mean': 0.5,
            'std': 0.5,
            'augmentation': False,
        },
        'loader': {
            'type': 'RecDataLoader',
            'batch_size': 4,
            'shuffle': False,
            'num_workers': 1,
            'collate_fn': {
                'type': 'RecCollateFn',
                'img_w': 120
            }
        }
    }
}

`

## 在执行上述训练命令后，会出现随机错误。日志如下（我在您的每一行代码下加了print，便于定位错误的位置）：
`  train loop: 1

0
 [INFO] end zero_grad 
 [INFO] end forward 
 [INFO] loss: tensor(6.5425, grad_fn=<MeanBackward0>)
 [INFO] end loss
  [INFO] end backward
 [INFO] end clip_grad_norm_
 [INFO] end optimizer step
1
 [INFO] end zero_grad
 [INFO] end forward
 [INFO] loss: tensor(nan, grad_fn=<MeanBackward0>)
 [INFO] end loss
 [INFO] end backward
 [INFO] end clip_grad_norm_
 [INFO] end optimizer step
2
 [INFO] end zero_grad
 [INFO] end forward
 [INFO] loss: tensor(nan, grad_fn=<MeanBackward0>)
 [INFO] end loss
 [INFO] end backward
 [INFO] end clip_grad_norm_
 [INFO] end optimizer step
3
 [INFO] end zero_grad
 [INFO] end forward
 [INFO] loss: tensor(nan, grad_fn=<MeanBackward0>)
 [INFO] end loss
 [INFO] end backward
 [INFO] end clip_grad_norm_
 [INFO] end optimizer step
4
 [INFO] end zero_grad
 [INFO] end forward
 [INFO] loss: tensor(nan, grad_fn=<MeanBackward0>)
 [INFO] end loss
 [INFO] end backward
 [INFO] end clip_grad_norm_
 [INFO] end optimizer step
5
 [INFO] end zero_grad
 [INFO] end forward
 [INFO] loss: tensor(nan, grad_fn=<MeanBackward0>)
 [INFO] end loss
Segmentation fault (core dumped)`
## 希望得到您的回复和指导，非常感谢！

**Comment by WenmuZhou** [comment]
[这一行](https://github.com/WenmuZhou/PytorchOCR/blob/master/config/rec_train_config.py#L58)的n_class也要做对应的修改，具体为字符数+1
